rootfs/etc/profile.d/atmos.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+function _configure_atmos_base_path() {
+	# Leave $ATMOS_BASE_PATH alone if it is already set
+	if [[ -n $ATMOS_BASE_PATH ]]; then
+		if [[ $SHLVL == 1 ]]; then
```

Comment:
Can't this be part of the same `if` conditional? i.e. `[[ -n $ATMOS_BASE_PATH ]] && [[ $SHLVL == 1 ]]` ? 

Also I think there should be an inline comment explaining why we are specifically testing for the number of shell levels.

Reply:
There is a classic example of a bad comment:
```
x = x + 1   // Add 1 to x
```
I didn't put a comment in about `SHLVL` because it seems to me it would be too much like that kind of bad comment. 

```bash
# Only output status in top level shell
if [[ $SHLVL == 1 ]]; then
```

The more useful comment would be about the bigger architectural design choice, but that would be long and does not belong in this bit of code. Since you asked, I will explain here in this comment.

The startup scripts (everything in `profile.d`) are responsible for a lot of the behavior of the interactive shell. Every time we spawn an interactive subshell, such as with `assume-role`, we run all these scripts again in the subshell to set it up. However, the startup messages are intended to expose configuration details of the Docker container or, if you will, the Geodesic session, more generally, and we do not want to repeat them when starting up a subshell with `assume-role` because (1) at that point they are not news and (2) its unnecessarily distracting. If you look around, you will find test for `$SHLVL == 1` all over the scripts in `profile.d`.